packages/ui/src/containers/UsernameSignin/index.tsx

```diff
@@ -0,0 +1,164 @@
+/**
+ * TODO:
+ * 1. API redesign handle api error and loading status globally in PageContext
+ * 2. Input field validation, should move the validation rule to the input field scope
+ * 3. Forgot password URL
+ * 4. read terms of use settings from SignInExperience Settings
+ */
+
+import { LogtoErrorI18nKey } from '@logto/phrases';
+import classNames from 'classnames';
+import React, { FC, useState, useCallback, useEffect, useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { signInBasic } from '@/apis/sign-in';
+import Button from '@/components/Button';
+import { ErrorType } from '@/components/ErrorMessage';
+import Input from '@/components/Input';
+import PasswordInput from '@/components/Input/PasswordInput';
+import TermsOfUse from '@/components/TermsOfUse';
+import TextLink from '@/components/TextLink';
+import PageContext from '@/hooks/page-context';
+import useApi from '@/hooks/use-api';
+
+import * as styles from './index.module.scss';
+
+type FieldState = {
+  username: string;
+  password: string;
+  termsAgreement: boolean;
+};
+
+type ErrorState = {
+  [key in keyof FieldState]?: ErrorType;
+};
+
+const defaultState: FieldState = {
+  username: '',
+  password: '',
+  termsAgreement: false,
+};
+
+const UsernameSignin: FC = () => {
+  const { t, i18n } = useTranslation();
+  const [fieldState, setFieldState] = useState<FieldState>(defaultState);
+  const [errors, setErrors] = useState<ErrorState>({});
```

Comment:
nit: Rename it to `fieldErrors`? As `errors` and `error` from useApi are a bit confusing

Reply:
update PTAL @demonzoo